### PR TITLE
NIFI-7799: Relogin with Kerberos on connect exception in DBCPConnectionPool

### PR DESCRIPTION
**Pull request by mattyb149**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

When using Kerberos and a load balancer to connect to a database, it is possible for DBCPConnectionPool to return a connection to the load balancer but logged in with the wrong ticket for the target database (this happens in High Availability scenarios). This PR tries to relogin to Kerberos if an exception occurs during connect.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-mock/pom.xml [hunk 1]
```diff
@@ -58,6 +58,11 @@
             <artifactId>nifi-data-provenance-utils</artifactId>
             <version>1.13.0-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-kerberos-credentials-service-api</artifactId>
+            <version>1.13.0-SNAPSHOT</version>
+        </dependency>
```
Comment (bbende) [inline nifi-mock/pom.xml:65]: I lean towards `nifi-mock `only really providing mocks for things in `nifi-api `and `nifi-framework-api`, but not really for extensions related things like a controller service api. 

I can see the need to have shared mock impls for controller services though, maybe a new test utils module under `nifi-nar-bundles/nifi-extension-utils`?

Comment (mattyb149) [reply nifi-mock/pom.xml:65]: That makes sense, will move this into a new module.

**Comment by mattyb149** [comment]
I force-pushed to make cherry-picking easier, all I did for the "second commit" was to move the Mock object into its own module under nifi-extension-utils as you suggested.

**Comment by bbende** [comment]
+1 changes look good, going to merge, thanks!

**Pull request merged**